code/modules/fishing/sources/_fish_source.dm

```diff
@@ -226,6 +226,9 @@ GLOBAL_LIST_INIT(specific_fish_icons, generate_specific_fish_icons())
 	if(isnull(reward_path))
 		return null
 	if(reward_path in fish_counts) // This is limited count result
+		//Somehow, we're trying to spawn an expended reward.
+		if(fish_counts[reward_path] <= 0)
+			return null
```

Comment:
if you know it's gonna be a number do this instead
```suggestion
	if(!isnull(fish_counts[reward_path])) // This is limited count result
		//Somehow, we're trying to spawn an expended reward.
		if(fish_counts[reward_path] <= 0)
			return null
```
in is n^2 assoc is log n. also it's more clear what's going on

Reply:
(for clarity this code is cold more a best practice sort of thing)